contracts/WildcardSteward_v2.sol

```diff
@@ -214,33 +220,56 @@ contract WildcardSteward_v2 is Initializable {
         }
     }
 
-    function addTokenGenerationRateToExistingTokens(
-        uint256[] memory tokens,
-        uint256[] memory _tokenGenerationRate
-    ) internal {
-        assert(tokens.length == _tokenGenerationRate.length);
+    function fixTotalPatronTokenGenerationRate(uint256[] memory tokens)
+        public
+        onlyAdmin
+    {
         for (uint8 i = 0; i < tokens.length; ++i) {
-            assert(tokenGenerationRate[tokens[i]] == 0);
+            uint256 tokenId = tokens[i];
+            address currentOwner = currentPatron[tokenId];
 
-            tokenGenerationRate[tokens[i]] = _tokenGenerationRate[i];
-        }
-    }
+            // NOTE: for this upgrade we make sure no tokens are foreclosed, or close to foreclosing
+            // _collectPatronage(tokenId);
 
-    function setMintManager(address _mintManager) public {
-        require(
-            address(mintManager) == address(0),
-            "Only set on initialisation"
-        );
-        mintManager = MintManager_v2(_mintManager);
-    }
+            // uint256 previousTokenCollection = timeLastCollected[tokenId];
 
-    function updateToV2(
-        address _mintManager,
-        uint256[] memory tokens,
-        uint256[] memory _tokenGenerationRate
-    ) public {
-        setMintManager(_mintManager);
-        addTokenGenerationRateToExistingTokens(tokens, _tokenGenerationRate);
+            uint256 collection = price[tokenId]
+                .mul(now.sub(timeLastCollected[tokenId]))
+                .mul(patronageNumerator[tokenId])
+                .div(1000000000000)
+                .div(365 days);
+
+            // uint256 timeSinceLastMint = now.sub(timeLastCollected[tokenId]);
+
+            timeLastCollected[tokenId] = now;
+            timeLastCollectedPatron[currentOwner] = now;
+            currentCollected[tokenId] = currentCollected[tokenId].add(
+                collection
+            );
+            deposit[currentOwner] = deposit[currentOwner].sub(
+                patronageOwedPatron(currentOwner)
+            );
+
+            totalCollected[tokenId] = totalCollected[tokenId].add(collection);
+
+            benefactorFunds[benefactors[tokenId]] = benefactorFunds[benefactors[tokenId]]
+                .add(collection);
+
+            emit CollectPatronage(
+                tokenId,
+                currentOwner,
+                deposit[currentOwner],
+                collection
+            );
+
+            _collectLoyaltyPatron(
+                currentOwner,
+                now.sub(timeLastCollected[tokenId])
+            );
+
+            totalPatronTokenGenerationRate[currentOwner] = totalPatronTokenGenerationRate[currentOwner]
+                .add(11574074074074);
```

Comment:
```suggestion
                 // 11574074074074 = 10^18 / 86400 This is just less (rounded down) than one token a day.
                 //       - this can be done since all tokens have the exact same tokenGenerationRate - and hardcoding saves gas.
                .add(11574074074074);
```